hushline/routes.py

```diff
@@ -417,16 +417,69 @@ def get_directory_users() -> list[User]:
             .all()
         )
 
+    def usersPerPage() -> int:
+        return 50
+
     @app.route("/directory")
     def directory() -> Response | str:
         logged_in = "user_id" in session
-        return render_template("directory.html", users=get_directory_users(), logged_in=logged_in)
+        isViewAllRoute = request.args.get('viewing') == 'all'
+        page = int(request.args.get('page', 0))
+        offset = page  * usersPerPage() - usersPerPage()
+        return render_template(
+            "directory.html",
+            users=get_dynamic_users(offset=offset, is_not_verified_only=isViewAllRoute),
+            logged_in=logged_in,
+            isViewAllRoute=isViewAllRoute,
+            usersPerPage=usersPerPage()
+        )
 
     @app.route("/directory/get-session-user.json")
     def session_user() -> dict[str, bool]:
         logged_in = "user_id" in session
         return {"logged_in": logged_in}
 
+    def get_dynamic_users(
+            count:int=usersPerPage(),
+            offset:int=0,
+            is_not_verified_only:bool = False
+        ) -> list[User]:
+        if is_not_verified_only:
+            return (
+                User.query.filter_by(show_in_directory=True)
+                .order_by(User.is_admin.desc(), User.display_name.asc())
+                .limit(count)
+                .offset(offset)
+                .all()
+            )
+        return (
+            User.query.filter_by(show_in_directory=True, is_verified=True)
+            .order_by(User.is_admin.desc(), User.display_name.asc())
+            .limit(count)
+            .offset(offset)
+            .all()
+        )
+
+    @app.route("/directory/pagination-users.json")
+    def pagination_users() -> list[dict[str, str]]:
+        count = request.args.get('count')
+        offset = request.args.get('offset')
```

Comment:
Do these need defaults? Otherwise I think we may not get the intended value if the parameters aren't in the args.
```suggestion
        count = request.args.get('count', usersPerPage())
        offset = request.args.get('offset', 0)
```